src/App.tsx

```diff
@@ -190,6 +192,10 @@ export default function App() {
 					https://www.boisestate.edu/coen-cs/community/cs481-senior-design-project/
 				</a>
 			</div>
+			<div id={"help-popup"} style={help ? {visibility: 'visible'} : {visibility: 'hidden'}}>
```

Comment:
probably better to use conditional rendering instead of styles here

Reply:
I'm playing around with this now. Be aware it introduces some perhaps unexpectedly difficult changes.

Since our popups open either in or above the viz area, either:
- the RenderView also needs to be conditionally rendered (in which case, toggling back and forth resets the camera position to its starting location)
- the RenderView isn't conditionally rendered, but all other popups are (in which case, relative positioning of the parent container #viz-area required as a base for child popups' absolute positioning breaks our raycast entirely; not sure exactly why, but no markers are selectable)
- popups are conditionally rendered in a popup container above RenderView, say z-index: 1 (in which case, our RenderView camera controls are blocked by the container since the mouseover that enables them doesn't reach RenderView below; in the current implementation, this is avoided with hidden visibility on all popup containers, which doesn't interfere with mouseover regardless of z-index)

Reply:
Test code:
```typescript
<div id={"viz-area"}>
    {!(error||sdpInfo||menu) ?
        <RenderView frame={frame} markerData={markerFileData} forceData={forceFileData}
                    selectedMarkers={selectedMarkers} setSelectedMarkers={setSelectedMarkers}
        />
        :<></>
    }
    {error ?
        <div id={"error-popup"}>Error: {error.message}</div>
        :<></>
    }
    {sdpInfo ?
        <div id={"sdp-info-popup"}>
            {`This website was created for a Boise State University
            Computer Science Senior Design Project by
            
            Colin Reeder
            Connor Jackson
            Cory Tomlinson
            Javier Trejo
            William Kenny
            
            For information about sponsoring a project, go to
            `}
            <a href={"https://www.boisestate.edu/coen-cs/community/cs481-senior-design-project/"} target={'_blank'}>
                https://www.boisestate.edu/coen-cs/community/cs481-senior-design-project/
            </a>
        </div>
        :<></>
    }
    {menu ?
        <div id={"main-menu-popup"}>
            <div id={"popup-image-container"}>
                <img id={"camera-controls-img"} src={cameraControlsImg} alt={"camera controls help image"}
                     style={controlsHelpImgNum === 0 ? {display: "block"} : {display: "none"}}/>
                <img id={"selection-controls-img"} src={selectionControlsImg} alt={"selection controls help image"}
                     style={controlsHelpImgNum === 1 ? {display: "block"} : {display: "none"}}/>
            </div>
            <dl id={"main-menu-options"}>
                <dt>Controls Help</dt>
                <dd id={"camera-controls"} onMouseOver={() => setControlsHelpImgNum(0)}>camera</dd>
                <dd id={"selection-controls"} onMouseOver={() => setControlsHelpImgNum(1)}>selection</dd>
            </dl>
        </div>
        :<></>
    }
</div>
```
```scss
#viz-area {
   grid-row: 2/4;
   grid-column: 1/3;
   // ---------- render view ----------
   background-color: $vizAreaBGColor;
   overflow: hidden;
   #render-view {
      height: 100%;
      width: 100%;
      position: absolute;
   }
   // ---------- popups ----------
   $popupPadding: 15px;
   position: relative; //base for child popups' absolute positioning
   font-size: 9pt; //for child popups to inherit
   #error-popup {
      position: absolute;
      padding: $popupPadding;
      width: calc(100% - 2*$popupPadding);
      height: calc(100% - 2*$popupPadding);
      overflow: auto;
      background-color: $errorPopupBGColor;
      color: $errorPopupTextColor;
      font-family: 'monospace','sans-serif';
      font-size: inherit;
      white-space: pre;
      cursor: text;
   }
   #main-menu-popup {
      position: absolute;
      padding: $popupPadding;
      width: calc(100% - 2*$popupPadding);
      height: calc(100% - 2*$popupPadding);
      display: flex;
      font-size: inherit;
      background-color: white;
      #popup-image-container {
         display: flex;
         align-items: center;
         overflow-y: auto;
         width: 100%; //maintain width even when no image is displayed
         img {
            width: 100%;
         }
      }
      #main-menu-options {
         $definitionTermWidth: 100px;
         border: solid 2px #444444;
         padding: 10px;
         > * {
            font-size: 9pt;
         }
         dt {
            float: left;
            clear: left;
            width: $definitionTermWidth;
            text-align: right;
            font-weight: bold;
            color: #025675;
            &::after {
               content: ":";
            }
         }
         dd {
            margin-left: calc($definitionTermWidth + 10px);
            &:hover {
               color: #9b9b0b;
            }
         }
      }
   }
   #sdp-info-popup {
      position: absolute;
      padding: $popupPadding;
      width: calc(100% - 2*$popupPadding);
      height: calc(100% - 2*$popupPadding);
      background-color: $sdpInfoPopupBGColor;
      color: $sdpInfoPopupTextColor;
      font-size: inherit;
      white-space: pre-line;
      display: flex;
      flex-direction: column; //display toggled between flex and none in App.tsx
      justify-content: center;
      align-items: center;
      text-align: center;
      a {
         text-decoration: none;
         color: $sdpLinkColor;
         font-size: inherit;
         font-weight: bold;
      }
   }
}
```